samza-azure/src/main/java/org/apache/samza/system/eventhub/SamzaEventHubClientManager.java

```diff
@@ -0,0 +1,101 @@
+* under the License.
+*/
+
+package org.apache.samza.system.eventhub;
+
+import com.microsoft.azure.eventhubs.EventHubClient;
+import com.microsoft.azure.servicebus.ClientConstants;
+import com.microsoft.azure.servicebus.ConnectionStringBuilder;
+import com.microsoft.azure.servicebus.RetryExponential;
+import com.microsoft.azure.servicebus.RetryPolicy;
+import com.microsoft.azure.servicebus.ServiceBusException;
+import org.apache.samza.SamzaException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+public class SamzaEventHubClientManager implements EventHubClientManager {
+  private static final Logger LOG = LoggerFactory.getLogger(SamzaEventHubClientManager.class.getName());
+
+  private static final String EVENTHUB_REMOTE_HOST_FORMAT = "%s.servicebus.windows.net";
+  private static final Duration MIN_RETRY_BACKOFF = Duration.ofMillis(100);
+  private static final Duration MAX_RETRY_BACKOFF = Duration.ofMillis(11000);
+  private static final int MAX_RETRY_COUNT = 100;
+  private static final String SAMZA_EVENTHUB_RETRY = "SAMZA_CONNECTOR_RETRY";
+
+  private EventHubClient eventHubClient;
+
+  private final String eventHubNamespace;
+  private final String entityPath;
+  private final String sasKeyName;
+  private final String sasKey;
+  private final RetryPolicy retryPolicy;
+
+  public SamzaEventHubClientManager(String eventHubNamespace, String entityPath, String sasKeyName, String sasKey) {
+    this(eventHubNamespace, entityPath, sasKeyName, sasKey,
+            new RetryExponential(MIN_RETRY_BACKOFF, MAX_RETRY_BACKOFF, MAX_RETRY_COUNT, SAMZA_EVENTHUB_RETRY));
+  }
+
+  public SamzaEventHubClientManager(String eventHubNamespace, String entityPath, String sasKeyName, String sasKey,
+                                    RetryPolicy retryPolicy) {
+    this.eventHubNamespace = eventHubNamespace;
+    this.entityPath = entityPath;
+    this.sasKeyName = sasKeyName;
+    this.sasKey = sasKey;
+    this.retryPolicy = retryPolicy;
+  }
+
+  public void init() {
+    String remoteHost = String.format(EVENTHUB_REMOTE_HOST_FORMAT, eventHubNamespace);
+    try {
+      ConnectionStringBuilder connectionStringBuilder =
+              new ConnectionStringBuilder(eventHubNamespace, entityPath, sasKeyName, sasKey);
+
+      eventHubClient = EventHubClient.createFromConnectionStringSync(connectionStringBuilder.toString(), retryPolicy);
+    } catch (IOException | ServiceBusException e) {
+      String msg = String.format("Creation of EventHub client failed for eventHub %s %s %s %s on remote host %s:%d",
+              entityPath, eventHubNamespace, sasKeyName, sasKey, remoteHost, ClientConstants.AMQPS_PORT);
+      LOG.error(msg, e);
+      throw new SamzaException(msg, e);
+    }
+  }
+
+  public EventHubClient getEventHubClient() {
+    return eventHubClient;
+  }
+
+
```

Comment:
Minor: Use override to explicitly indicate that it's overriding a super-type's method. Likewise for other overridden methods